abuild: rename makedepends_host virtual package

ollieparanoid opened the pull request:
With a recent change in apk [1], virtual packages of the same name will
upgrade each other. Adjust abuild to this by not using the same virtual
package name for two types of dependencies.

This fixes the way crosscompilers are built in postmarketOS [2], which
is essentially the same as running this on Alpine's gcc aport:
```
$ cd aports/main/gcc
$ C_TARGET_ARCH=armhf CTARGET=armv6-alpine-linux-musleabihf \
  BOOTSTRAP=nobuildbase CBUILDROOT=/ abuild -r
...
>>> gcc-armhf: Installing for host:
(1/24) Upgrading .makedepends-gcc-armhf (20190714.104731 -> 20190714.104741)
(2/24) Purging binutils-armhf (2.31.1-r2)
...
```

[1] https://github.com/alpinelinux/apk-tools/commit/37fbafcd928c466c82c892a7868d686d710e5d07
[2] https://gitlab.com/postmarketOS/pmaports/blob/master/cross/gcc-armhf/APKBUILD

Fixes: https://gitlab.alpinelinux.org/alpine/apk-tools/issues/10649


CC: @fabled

Commit:
abuild: rename makedepends_host virtual package

With a recent change in apk [1], virtual packages of the same name will
upgrade each other. Adjust abuild to this by not using the same virtual
package name for two types of dependencies.

This fixes the way crosscompilers are built in postmarketOS [2], which
is essentially the same as running this on Alpine's gcc aport:
$ cd aports/main/gcc
$ C_TARGET_ARCH=armhf CTARGET=armv6-alpine-linux-musleabihf \
  BOOTSTRAP=nobuildbase CBUILDROOT=/ abuild -r
...
>>> gcc-armhf: Installing for host:
(1/24) Upgrading .makedepends-gcc-armhf (20190714.104731 -> 20190714.104741)
(2/24) Purging binutils-armhf (2.31.1-r2)
...

[1] apk-tools.git 37fbafcd928c466c82c892a7868d686d710e5d07
    ("add: make virtual packages upgradeable (ref #9957)")
[2] https://gitlab.com/postmarketOS/pmaports/blob/master/cross/gcc-armhf/APKBUILD

Fixes: https://gitlab.alpinelinux.org/alpine/apk-tools/issues/10649

ncopa commented on abuild.in:
```diff
@@ -2424,13 +2424,13 @@ deps() {
 undeps() {
 	local _quiet="$@"
-	$SUDO_APK del $_quiet $apk_opt_wait .makedepends-$pkgname || :
+	$SUDO_APK del $_quiet $apk_opt_wait .makedepends-$pkgname .hostdepends-$pkgname || :
```
Should'nt the `.hostdepends-$pkgname` be removed within the `if [ -n "$CBUILDROOT" ]; then` and shouldn't it have a `--root "$CBUILDROOT"` option too?

ncopa commented on abuild.in:
```diff
@@ -2424,13 +2424,13 @@ deps() {
+	$SUDO_APK del $_quiet $apk_opt_wait .makedepends-$pkgname .hostdepends-$pkgname || :
 	if [ -n "$CBUILDROOT" ]; then
 		$SUDO_APK del $_quiet --root "$CBUILDROOT" --arch "$CTARGET_ARCH" $apk_opt_wait \
 			--no-scripts .makedepends-$pkgname || :
```
```suggestion
			--no-scripts .makedepends-$pkgname .hostdepends-$pkgname || :
```

ncopa reviewed (changes requested):
the `--root ...` option in `$SUDO_APK del` should match `$SUDO_APK add`

ollieparanoid commented:
Thanks for the review @ncopa!

You are right, the way I've changed `undeps()` by adding `.hostdepends-$pkgname` to the `$SUDO_APK del` line outside of the `if [ -n "$CBUILDROOT" ]; then` condition did not make much sense...

I've changed it to uninstall the `.hostdepends-$pkgname` only if `$CBUILDROOT` is set. Because that is the only way they get installed in `deps()`. So this should be the right way to do it. I've verified again that the gcc cross compiling use case is fixed with this patch.

Can you take another look at it?
